### DISCUSSION TITLE
Cant upgrade from the alpha version :(

### DISCUSSION THREAD

**Discussion by AndreySylka** [strapi/strapi#7618] [Legacy Topics]
Hello to all, 

I inherit project that use alpha version of strapi. I try to upgrade it to the latest version. 
1. set i try to upgrade to beta version following this documentation https://strapi.io/documentation/v3.x/migration-guide/migration-guide-alpha.26-to-beta.html#migrating-customized-plugin. I have done all steps up to migrating customized plugins

i try run the project but got this errors that i even  cant google 
> strapi build

Building your admin UI with development configuration ...

√ Webpack
  Compiled successfully in 10.03s

C:\Users\ansylka\Desktop\Repos\react\xom-apie-cms-app>npm run start

> strapi-app@0.1.0 start C:\Users\ansylka\Desktop\Repos\react\xom-apie-cms-app
> strapi start

[2020-08-26T19:49:47.908Z] debug ⛔️ Server wasn't able to start properly.
[2020-08-26T19:49:47.912Z] error TypeError: Cannot convert undefined or null to object
    at Function.values (<anonymous>)
    at Object.load (C:\Users\ansylka\Desktop\Repos\react\xom-apie-cms-app\node_modules\strapi-database\lib\connector-registry.js:17:39)
    at DatabaseManager.initialize (C:\Users\ansylka\Desktop\Repos\react\xom-apie-cms-app\node_modules\strapi-database\lib\database-manager.js:32:21)
    at Strapi.load (C:\Users\ansylka\Desktop\Repos\react\xom-apie-cms-app\node_modules\strapi\lib\Strapi.js:299:19)
    at async Strapi.start (C:\Users\ansylka\Desktop\Repos\react\xom-apie-cms-app\node_modules\strapi\lib\Strapi.js:158:9)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! strapi-app@0.1.0 start: `strapi start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the strapi-app@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ansylka\AppData\Roaming\npm-cache\_logs\2020-08-26T19_49_47_955Z-debug.log

C:\Users\ansylka\Desktop\Repos\react\xom-apie-cms-app>

**Comment by derrickmehaffy** [comment]
@AndreySylka to be quite frank about the migration from Alpha to the Stable release, I would very strongly suggest you start with a fresh project and rebuild the models first (not too much has changed in the model settings files but a bit has through various migration guides) and how you do that is up to you. I would suggest you generate all the models first using something `yarn strapi generate:api modelName blah:string` **without starting strapi** then start copying over the actual model settings files in `./api/*/models/*.settings.json`.

Once you have all the models constructed and setup properly you can start moving towards modifying the plugins how you need to, in which this process has changed entirely and as have the plugins so from here you will more or less be starting fresh to rebuild any custom code. I would suggest you thoroughly read through all of the new concept documentation to get familiar with how the internal API works for the follow:
- Config structure: https://strapi.io/documentation/v3.x/concepts/configurations.html
- Controllers: https://strapi.io/documentation/v3.x/concepts/controllers.html
- Services: https://strapi.io/documentation/v3.x/concepts/services.html
- Queries: https://strapi.io/documentation/v3.x/concepts/queries.html
- Middlewares: https://strapi.io/documentation/v3.x/concepts/middlewares.html
- Policies: https://strapi.io/documentation/v3.x/concepts/policies.html

And once you are familiar with that you can move onto the [extensions system](https://strapi.io/documentation/v3.x/concepts/customization.html#plugin-extensions) and customizing the admin (including the plugins): https://strapi.io/documentation/v3.x/concepts/customization.html#admin-extension

Node that if you need to customize the backend of the `strapi-admin` package you will be forced to use a local node module reference in your package.json (I would suggest avoiding that if possible as for every version update including minor ones you will be forced to review the changes to the package and manually apply them).

Simply following all the migration guides for the past 2 years worth of updates one by one you are going to hit a ton of problems and take probably 3 times as long to update vs just starting fresh.
